rust/src/go/pkg/nmstate/nmstate.go

```diff
@@ -0,0 +1,53 @@
+package nmstate
+
+// #cgo CFLAGS: -g -Wall
+// #cgo LDFLAGS: -L/usr/local/lib64 -lnmstate -Wl,-rpath=/usr/local/lib64
+// #include <nmstate.h>
+// #include <stdlib.h>
+import "C"
+import "fmt"
+
+func RetrieveNetState(flags uint) (string, error) {
```

Comment:
We can use bitmask at golang, https://play.golang.org/p/_bTfJBuonzs

```suggestion
type Flags uint

const None Flags = 0

const (
    KernelOnly = 1 << iota
)
func RetrieveNetStateWithFlags(flags Flags) (string, error) {
...
}

func RetrieveNetState() (string, error) {
    return RetrieveNetStateWithFlags(None)
}
```

Reply:
Done, I have introduced bitmask following the pattern you described. Thank you!